nncf/common/accuracy_aware_training/training_loop.py

```diff
@@ -0,0 +1,243 @@
+                 runner_cls=None):
+        self.adaptive_controller, accuracy_aware_config = self._get_adaptive_compression_ctrl(compression_controller,
+                                                                                              nncf_config)
+        runner_cls = AccuracyAwareTrainingRunner if runner_cls is None else runner_cls
+        self.runner = runner_cls(accuracy_aware_config)
+        if self.adaptive_controller is None:
+            raise RuntimeError('No compression algorithm supported by the accuracy-aware training '
+                               'runner was specified in the config')
+
+    def _get_adaptive_compression_ctrl(self, compression_controller, nncf_config):
+        adaptive_compression_controllers = self._adaptive_compression_controllers()
+        compression_configs = nncf_config.get('compression', {})
+        if isinstance(compression_configs, list):
+            comp_algorithm_params_dict = {compression_config['algorithm']: compression_config
+                                          for compression_config in compression_configs}
+        else:
+            comp_algorithm_params_dict = {compression_configs['algorithm']: compression_configs}
+
+        if not isinstance(compression_controller, CompositeCompressionAlgorithmController):
+            for algo_name, ctrl_type in adaptive_compression_controllers.items():
+                if isinstance(compression_controller, ctrl_type):
+                    acc_aware_config = comp_algorithm_params_dict[algo_name].get('accuracy_aware_training', None)
+                    return compression_controller, acc_aware_config
+        if isinstance(compression_controller, CompositeCompressionAlgorithmController):
+            for controller in compression_controller.child_ctrls:
+                for algo_name, ctrl_type in adaptive_compression_controllers.items():
+                    if isinstance(controller, ctrl_type):
+                        acc_aware_config = comp_algorithm_params_dict[algo_name].get('accuracy_aware_training', None)
+                        return controller, acc_aware_config
+        raise RuntimeError('No compression algorithm that supports adaptive compression '
+                           'accuracy-aware training was specified')
+
+    def _adaptive_compression_controllers(self):
+        def remove_registry_prefix(algo_name):
+            for prefix in ('pt_', 'tf_'):
+                if algo_name.startswith(prefix):
+                    return algo_name[len(prefix):]
+            raise RuntimeError('Compression algorithm names in the adaptive controllers '
+                               'registry should be prefixed with "pt_" or "tf_" depending on the '
+                               'backend framework')
+        return {remove_registry_prefix(algo_name): controller_cls for algo_name, controller_cls in
+                ADAPTIVE_COMPRESSION_CONTROLLERS.registry_dict.items()}
+
+    # pylint: disable=E1101
```

Comment:
Is there a way to fix this? Or is it the pylint issue?

Reply:
done